gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -1184,17 +1184,16 @@ async fn test_gateway_executes_swaps_between_connected_federations() -> anyhow::
             assert_matches!(claimed, LnReceiveState::Claimed);
             assert_eq!(client2.get_balance().await, invoice_amt);
 
-            // FIXME: https://github.com/fedimint/fedimint/issues/3672
-            // This assertion intermittently fails and is a blocker on the CI pipeline
-            // we need to investigate further why the gateway balance may not be updating
-            //
+            // Give the gateway a chance to update it's balances
+            sleep(Duration::from_secs(5)).await;
```

Comment:
You could speed this up by polling the balance for 5 seconds, and fail the test if it doesn't update in 5 seconds, but continue if it does.

Reply:
updated

found it non-intuitive that we have to do
```rs
if post_balances[0] == pre_balances[0] && post_balances[1] == pre_balances[1] { // error balances not updated }
```
and not
```rs
if post_balances == pre_balances { // error balances not updated }
```
because we are watching perf and latencies in two distinct federations